client/qr-worker.js

```diff
@@ -0,0 +1,13 @@
+import QrCode from 'qrcode-reader';
+
+self.onmessage = ({ data }) => {
+  let qr = new QrCode();
+  qr.callback = function(error, result) {
+    if(error) {
+      self.postMessage({ error });
+      return;
+    }
+    self.postMessage({result: result.result});
+  }
+  qr.decode(data);
+}
```

Comment:
This is part of the code that students will end up writing -- it's a pretty simple pattern for single-message workers. Maybe we should explicitly `.terminate()` after this is all done, just to make the point that a new one is instantiated on each button push (and this is fine!)